src/corecel/data/detail/LdgIteratorImpl.hh

```diff
@@ -28,7 +28,7 @@ struct LdgLoader
                   "Only const arithmetic types are supported by __ldg");
     using value_type = T;
     using pointer = std::add_pointer_t<value_type>;
-    using reference = value_type;
+    using reference = std::remove_const_t<value_type>;
```

Comment:
Shouldn't `value_type` have the `remove_const_t` and only `pointer` should have `T const*`?